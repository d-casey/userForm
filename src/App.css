.app-header {
    flex: 1;
    display: flex;
    text-align: center;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 3.5em;
    margin: 0 auto;
    max-width: 1170px;
}

.checkbox {
  flex: 1;
  display: flex;
  flex-direction: row;
  padding: 5px;
}

.checkbox-input {
  width: auto;
}

fieldset {
  border: none;
}

ul {
  list-style-type: none;
}

* {
  box-sizing: border-box;
}

body {
  padding: 1rem;
}

.error {
  color: #d33f3f
}

.main-container {
  max-width: 1170px;
  margin-left: auto;
  margin-right: auto;
  padding: 1em;
  margin-top: 5em;
}
@media (min-width: 600px) {
  .main-container {
    padding: 0;
  }
}

h3,
ul {
  margin: 0;
}

h3 {
  margin-bottom: 1rem;
}

input:focus,
textarea:focus {
  outline: 1px solid #0a0d36;
}

input,
textarea,
button {
  width: 100%;
  border: 1px solid #000;
}

.wrapper {
  box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.2);
}
.wrapper > * {
  padding: 1em;
}
@media (min-width: 700px) {
  .wrapper {
    display: grid;
    grid-template-columns: 1fr 2fr;
  }
  .wrapper > * {
    padding: 2em 2em;
  }
}

ul {
  list-style: none;
  padding: 0;
}

.sidebar {
  background: #0d66e5;
  color: #fff;
}

.form-container {
  background: #fff;
}

form {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
}
form label {
  display: block;
}
form p {
  margin: 0;
}

.full-width {
  grid-column: 1 / 3;
}

.right {
  grid-column: 2 / 3;
}

button,
input,
textarea {
  padding: 1em;
}

button {
  background: lightgrey;
  width: 100%;
  border: 0;
}
button:hover, button:focus {
  background: #0d66e5;
  outline: 0;
  color: white;
}


.progress-bar-flex-container {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  padding: 0;
  margin-bottom: 1em;
  list-style: none;
}

.progress-bar-flex-item {
  height: 30px;
  margin-top: 10px;
  line-height: 30px;
  color: #000;
  text-align: center;
  flex-grow: 1;
  border: 1px solid #000;
}

.active-item {
  background: #0d66e5;
  color: white;
}

.success {
  color: #0dbf73;
  text-align: center;
  font-weight: bold;
}
