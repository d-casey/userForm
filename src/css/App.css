fieldset {
  border: none;
}

ul {
  list-style-type: none;
}

* {
  box-sizing: border-box;
}

body {
  padding: 1rem;
}

.error {
  color: #d33f3f
}

h3,
ul {
  margin: 0;
}

h3 {
  margin-bottom: 1rem;
}

input:focus,
textarea:focus {
  outline: 1px solid #0a0d36;
}

input,
textarea,
button {
  width: 100%;
  border: 1px solid #000;
}

ul {
  list-style: none;
  padding: 0;
}

form {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
}
form label {
  display: block;
}
form p {
  margin: 0;
}

.right {
  grid-column: 2 / 3;
}

button,
input,
textarea {
  padding: 1em;
}

button {
  background: lightgrey;
  width: 100%;
  border: 0;
}
button:hover, button:focus {
  background: #0d66e5;
  outline: 0;
  color: white;
}

.form-success {
  color: #0dbf73;
  text-align: center;
  font-weight: bold;
}
