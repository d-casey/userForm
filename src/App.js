import React from 'react'
import Header from './components/Header'
import Main from './containers/Main'
import './App.css'

function App() {
  return (
    <div className="App">
      <Header />
      <Main />
    </div>
  )
}

export default App
