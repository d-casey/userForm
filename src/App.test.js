import React from 'react'
import App from './App'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

const mockStore = configureStore({})
let store
let component
let initialState = {
  app: {
    stage: 0,
    formCompleted: false
  },
  newUserForm: {
    name: "",
    role: "",
    email: "",
    password: "",
    isSubmitted: false
  },
  privacyForm : {
    isSubmitted: false,
    receive_updates: false,
    receive_communication: false,
  }
}

const getWrapper = () => {
  store = mockStore(initialState)
  component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  )
}

describe('<App /> container', () => {
  beforeEach(() => {
    getWrapper()
  })

  it('renders the content as expected', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
