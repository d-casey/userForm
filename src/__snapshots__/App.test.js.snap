// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> container renders the content as expected 1`] = `
<div
  className="App"
>
  <header
    className="app-header"
  >
    <img
      alt="trayLogo"
      src="logo.png"
    />
  </header>
  <div
    className="main-container"
  >
    <div
      className="progress-bar"
    >
      <ul
        className="progress-bar-flex-container"
      >
        <li
          className="progress-bar-flex-item active-item"
        >
          User
        </li>
        <li
          className="progress-bar-flex-item"
        >
          Privacy
        </li>
        <li
          className="progress-bar-flex-item"
        >
          Done
        </li>
      </ul>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="sidebar"
      >
        <h3>
          Sign Up Form
        </h3>
        <p>
          Required fiels are marked with an *
        </p>
        <p>
          Passwords must be:
        </p>
        <ul>
          <li>
            At least 10 characters long
          </li>
          <li>
            Contain at least one number
          </li>
          <li>
            Contain at least one upper case letter
          </li>
          <li>
            Contain at least one lower case letter
          </li>
        </ul>
      </div>
      <div
        className="form"
      >
        <form
          data-testid="user-form"
          onSubmit={[Function]}
        >
          <fieldset
            className="full-width-imput"
          >
            <legend>
              Name
              : 
              *
               
              <span
                className="error"
              />
            </legend>
            <input
              data-testid="user_name"
              defaultValue=""
              name="name"
              type="text"
            />
          </fieldset>
          <fieldset
            className="full-width-imput"
          >
            <legend>
              Role
              : 
              
               
              <span
                className="error"
              />
            </legend>
            <input
              data-testid="user_role"
              defaultValue=""
              name="role"
              type="text"
            />
          </fieldset>
          <fieldset
            className="full-width-imput"
          >
            <legend>
              Email
              : 
              *
               
              <span
                className="error"
              />
            </legend>
            <input
              data-testid="user_email"
              defaultValue=""
              name="email"
              type="email"
            />
          </fieldset>
          <fieldset
            className="full-width-imput"
          >
            <legend>
              Password
              : 
              *
               
              <span
                className="error"
              />
            </legend>
            <input
              data-testid="user_password"
              name="password"
              type="password"
            />
          </fieldset>
          <button
            className="right"
            data-testid="user-form-submit-button"
            type="submit"
          >
            Next
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
